### DISCUSSION TITLE
Drivers/Vpython visualizations/examples?

### DISCUSSION THREAD

**Discussion by DarkSky7** [cosinekitty/astronomy#310] [General]
I looked at demo/python/astronomy.py - huge piece of code, but no main.  Where can I find test driver examples?  Are there plans for Vpython visualizations or existing examples to read?

**Comment by cosinekitty** [comment]
Hi @DarkSky7 and welcome to the project!

If you are looking for the unit tests to see how the Python version of Astronomy Engine is validated, there is a bash script [unit_test_python](https://github.com/cosinekitty/astronomy/blob/master/generate/unit_test_python) which runs pylint and mypy, then invokes the main test script [test.py](https://github.com/cosinekitty/astronomy/blob/master/generate/test.py). It also generates a large output file exercising the core functionality for comparing the calculations with the other programming language implementations for consistency.

If you are looking for sample programs, there is a [collection of demos](https://github.com/cosinekitty/astronomy/tree/master/demo/python) that are all console-mode programs.

About VPython, I had to look that up... it is new to me. It does look very interesting! So no, I didn't have any plans for that. I'm trying to understand if VPython is open source, and if so, where I can find the source code.

**Reply by DarkSky7** [reply]
Vulkan might do the trick too; it's newer and faster than OpenGL (upon which Vpython 'rides').  On the examples, I was looking for something 'simple' (a lot less than 10k lines) that I could learn from, maybe specify a UTC datetime and get back the state vectors of solar system objects - sun, planets, planetessimals, moons, major asteroids, comets, ... (whatever I can get).

**Comment by DarkSky7** [comment]
It's been around a long time, I saw nothing about licensing on:
https://vpython.org/index.html
https://vpython.org/contents/developers.html says it's open source.
